core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -587,6 +568,40 @@ private void determineGlobalShadowOrder() {
         }
     }
 
+    private void collectEntityAndProblemFactClasses() {
```

Comment:
Wouldn't it be nicer if this method were a stateless function, returning the result instead of setting a field on the object?
This approach has many benefits - such as allowing for testing methods in isolation and avoiding side effects.